app/models/service_template.rb

```diff
@@ -69,6 +69,65 @@ class ServiceTemplate < ApplicationRecord
   scope :with_existent_service_template_catalog_id, ->         { where.not(:service_template_catalog_id => nil) }
   scope :displayed,                                 ->         { where(:display => true) }
 
+  def self.catalog_item_types
+    CATALOG_ITEM_TYPES.each.with_object({}) do |(key, description), hash|
+      method_name = "#{key}_provider_present?".to_sym
+      hash[key] = { :description => description,
+                    :display     => ServiceTemplate.respond_to?(method_name) ? ServiceTemplate.send(method_name) : false }
+    end
+  end
+
+  def self.amazon_provider_present?
+    ExtManagementSystem.where(:type => "ManageIQ::Providers::Amazon::CloudManager").any?
+  end
+
+  def self.azure_provider_present?
+    ExtManagementSystem.where(:type => "ManageIQ::Providers::Azure::CloudManager").any?
+  end
+
+  def self.generic_provider_present?
+    true
+  end
+
+  def self.generic_orchestration_provider_present?
+    orchestration_providers = OrchestrationTemplate.distinct.pluck(:type).collect do |otype|
+      otype.constantize.eligible_managers.collect(&:type)
```

Comment:
`#eligible_managers` has `Rbac` (https://github.com/ManageIQ/manageiq/blob/master/app/models/orchestration_template.rb#L127). Should we check the same for other `provider_present?` in Ln 104, 108 etc?

Reply:
@bzwei 
https://github.com/ManageIQ/manageiq/blob/d66dbccded14bee7d2f5f379c9b6270ca7b4e31a/app/models/orchestration_template.rb#L127
Does not accept a User object so its most probably relying on the User.current_user being set in the UI controller based on the currently logged in user so it should do the right thing here. We dont have to pass the user object.

Reply:
@bzwei I checked with @kbrock and his suggestion was the RBAC calls might slow this down a bit. So for the first round can we just use the ActiveModel queries and then we can make the RBAC changes.

Reply:
1. you can pass in `:user` into the `filtered()` call
2. Rbac slows everything down. It is faster to just have no security.
3. The goal is to use rbac in your query along with other sql filtering (e.g. `with_eligible_manager_types`)

Does this work just calling the scope directly?

```ruby
Rbac::Filterer.filtered(ExtManagementSystem.with_eligible_manager_types(eligible_manager_types))
```